libs/core/langchain_core/language_models/llms.py

```diff
@@ -164,7 +168,11 @@ async def aget_prompts(
     missing_prompts = []
     missing_prompt_idxs = []
     existing_prompts = {}
-    llm_cache = get_llm_cache()
+    cache = params["cache"]
```

Comment:
let's expand the function signature to accept `cache` explicitly -- it should not be in `params` since `params` are used to construct the llm_string which is the cache key